dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/DockerRegistryAPIHelper.java

```diff
@@ -48,6 +64,139 @@
+
+        // Get token with pull access to use for subsequent Docker Registry HTTP API V2 calls
+        Optional<String> token = DockerRegistryAPIHelper.getDockerToken(registry.getDockerPath(), repo);
+        if (token.isEmpty()) {
+            LOG.error("Could not retrieve token for {} repository {}", registry.getFriendlyName(), repo);
+            return images;
+        }
+
+        // Get manifest for image. There are two types of manifests: image manifest and manifest list. Manifest list is used for multi-arch images.
+        Optional<Response> manifestResponse = DockerRegistryAPIHelper.getDockerManifest(token.get(), registry.getDockerPath(), repo, specifierName);
+        if (manifestResponse.isEmpty()) {
+            LOG.error("Could not retrieve manifest for {}", imageNameMessage);
+            return images;
+        }
+
+        // Check what type of manifest was returned and whether the image is multi-arch
+        String contentTypeHeader = manifestResponse.get().headers().get(HttpHeaders.CONTENT_TYPE);
+        if (contentTypeHeader == null) {
+            LOG.error("Could not retrieve Content-Type header from manifest response for {}", imageNameMessage);
+            return images;
+        }
+
+        String contentType = contentTypeHeader;
+        Reader manifestJson = manifestResponse.get().body().charStream();
+        String digest;
+
+        if (contentType.equals(DockerRegistryAPIHelper.DOCKER_V2_IMAGE_MANIFEST_MEDIA_TYPE) || contentType.equals(DockerRegistryAPIHelper.OCI_IMAGE_MANIFEST_MEDIA_TYPE)) {
+            DockerImageManifest imageManifest = GSON.fromJson(manifestJson, DockerImageManifest.class);
+
+            // Check that the image manifest is using schema version 2 because schema version 1 is deprecated and the JSON response is
+            // formatted differently, thus requiring some pre-processing before calculating the image digest
+            if (imageManifest.getSchemaVersion() != 2) {
+                LOG.error("The image manifest for {} is using schema version {}, not schema version 2", imageNameMessage, imageManifest.getSchemaVersion());
+                return images;
+            }
+
+            // The manifest response may include a Docker-Content-Digest header, which is the digest of the image
+            String digestHeader = manifestResponse.get().headers().get("docker-content-digest");
+            if (digestHeader == null) {
+                // Manually calculate the digest if not given in the header
+                digest = DockerRegistryAPIHelper.calculateDockerImageDigest(manifestResponse.get());
+
+                if (digest.isEmpty()) {
+                    LOG.error("Could not calculate digest for {}", imageNameMessage);
+                    return images;
+                }
+            } else {
+                digest = digestHeader.split(":")[1];
```

Comment:
All optional:

A comment with the format of `digestHeader` would be nice -- I was trying to visualize what it's splitting.

Is it `sha256:<digest>`? If yes, although it's probably overkill, you could even use the `sha256` portion of the string on line 130, in case they ever return a different SHA type. If not, never mind.